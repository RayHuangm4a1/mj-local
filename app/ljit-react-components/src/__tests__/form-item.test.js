import React from 'react';
import { mount, render, } from 'enzyme';
import FormItem, { ColumnTypeEnums, ColumnEnums, } from '../components/form-item';

jest.mock('../components/form', () => ({
	connectForm: function (Component) {
		return Component;
	},
}));

describe('Form Item', () => {
	const className = 'mock-class';
	const label = 'mock-label';
	const columnType = 'large';
	const labelColon = true;
	const labelRequired = true;
	const groupItemNames = ['test-item1', 'test-item2',];
	const groupItemConfigs = {
		'test-item1': {},
		'test-item2': {},
	};
	const form = {
		getFieldDecorator: (name, config) => (Component) => {
			return Component;
		},
	};

	afterEach(() => {
		jest.resetModules();
	});

	it('should renders correctly', () => {
		const wrapper = render(<FormItem
			className={className}
			label={label}
			form={form}
			itemName="test-item"
			columnType={columnType}
			labelColon={labelColon}
			labelRequired={labelRequired}
			noMargin
			groupItemNames={groupItemNames}
			groupItemConfigs={groupItemConfigs}
			isInputGroup={false}
			extraMessage="remind text"
		>
			mock-item
		</FormItem>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-form-label', () => {
		const wrapper = mount(<FormItem
			form={form}>
			mock-item
		</FormItem>);
		const wrapperFormItem = wrapper.find('.ljit-form-label').at(0);

		expect(wrapperFormItem.hasClass('ljit-form-label')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = mount(<FormItem
			className={className}
			form={form}>
			mock-item
		</FormItem>);
		const wrapperFormItem = wrapper.find(`.${className}`).at(0);

		expect(wrapperFormItem.hasClass(className)).toEqual(true);
	});

	it('should ljit-form-label--no-margin be selectable when pass noMargin', () => {
		const wrapper = mount(<FormItem
			form={form}
			noMargin>
			mock-item
		</FormItem>);
		const wrapperFormItem = wrapper.find('.ljit-form-label--no-margin').at(0);

		expect(wrapperFormItem.hasClass('ljit-form-label--no-margin')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const label = 'mock-label';
		const itemName = 'test-item';
		const columnType = 'large';
		const labelColon = true;
		const labelRequired = true;
		const groupItemNames = ['test-item1', 'test-item2',];
		const groupItemConfigs = {
			'test-item1': {},
			'test-item2': {},
		};
		const isInputGroup = false;
		const extraMessage = 'remind text';
		const wrapper = mount(<FormItem
			className={className}
			label={label}
			form={form}
			itemName={itemName}
			columnType={columnType}
			labelColon={labelColon}
			labelRequired={labelRequired}
			groupItemNames={groupItemNames}
			groupItemConfigs={groupItemConfigs}
			isInputGroup={isInputGroup}
			noMargin
			extraMessage={extraMessage}
		>
			mock-item
		</FormItem>);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().label).toBe(label);
		expect(wrapper.props().itemName).toBe(itemName);
		expect(wrapper.props().form).toBe(form);
		expect(wrapper.props().columnType).toBe(columnType);
		expect(wrapper.props().labelColon).toBe(labelColon);
		expect(wrapper.props().labelRequired).toBe(labelRequired);
		expect(wrapper.props().groupItemNames).toBe(groupItemNames);
		expect(wrapper.props().groupItemConfigs).toBe(groupItemConfigs);
		expect(wrapper.props().isInputGroup).toBe(isInputGroup);
		expect(wrapper.props().extraMessage).toEqual(extraMessage);
	});

	describe('Column Type Enums', () => {
		it('should contains large property', () => {
			const typeName = 'large';
			const columnType = 'LARGE';

			expect(ColumnTypeEnums[columnType]).toEqual(typeName);
		});

		it('should contains medium property', () => {
			const typeName = 'medium';
			const columnType = 'MEDIUM';

			expect(ColumnTypeEnums[columnType]).toEqual(typeName);
		});

		it('should contains small property', () => {
			const typeName = 'small';
			const columnType = 'SMALL';

			expect(ColumnTypeEnums[columnType]).toEqual(typeName);
		});
	});

	describe('Column Enums', () => {
		describe('Large', () => {
			let typeName;
			let columnEnum;

			beforeEach(() => {
				typeName = ColumnTypeEnums.LARGE;
				columnEnum = ColumnEnums[typeName];
			});
			afterEach(() => {
				typeName = undefined;
				columnEnum = undefined;
			});

			it('should be defined', () => {
				expect(columnEnum).toBeDefined();
			});

			it('should labelCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.xs.span).toBe(result);
			});

			it('should labelCol have 9 col in sm window size', () => {
				const result = 9;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.sm.span).toBe(result);
			});

			it('should wrapperCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.xs.span).toBe(result);
			});

			it('should wrapperCol have 15 col in sm window size', () => {
				const result = 15;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.sm.span).toBe(result);
			});
		});

		describe('Medium', () => {
			let typeName;
			let columnEnum;

			beforeEach(() => {
				typeName = ColumnTypeEnums.MEDIUM;
				columnEnum = ColumnEnums[typeName];
			});
			afterEach(() => {
				typeName = undefined;
				columnEnum = undefined;
			});

			it('should be defined', () => {
				expect(columnEnum).toBeDefined();
			});

			it('should labelCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.xs.span).toBe(result);
			});

			it('should labelCol have 6 col in sm window size', () => {
				const result = 6;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.sm.span).toBe(result);
			});

			it('should wrapperCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.xs.span).toBe(result);
			});

			it('should wrapperCol have 18 col in sm window size', () => {
				const result = 18;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.sm.span).toBe(result);
			});
		});

		describe('Small', () => {
			let typeName;
			let columnEnum;

			beforeEach(() => {
				typeName = ColumnTypeEnums.SMALL;
				columnEnum = ColumnEnums[typeName];
			});
			afterEach(() => {
				typeName = undefined;
				columnEnum = undefined;
			});

			it('should be defined', () => {
				expect(columnEnum).toBeDefined();
			});

			it('should labelCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.xs.span).toBe(result);
			});

			it('should labelCol have 4 col in sm window size', () => {
				const result = 4;

				expect(columnEnum.labelCol).toBeDefined();
				expect(columnEnum.labelCol.sm.span).toBe(result);
			});

			it('should wrapperCol have full width in xs window size', () => {
				const result = 24;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.xs.span).toBe(result);
			});

			it('should wrapperCol have 20 col in sm window size', () => {
				const result = 20;

				expect(columnEnum.wrapperCol).toBeDefined();
				expect(columnEnum.wrapperCol.sm.span).toBe(result);
			});
		});
	});
});
