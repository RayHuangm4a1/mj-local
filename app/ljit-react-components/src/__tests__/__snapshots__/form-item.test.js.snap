// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Item should renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-row ant-form-item ljit-form-label ljit-form-label--no-margin mock-class",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-9",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ant-form-item-required",
              "title": "mock-label",
            },
            "children": Array [
              Object {
                "data": "mock-label",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "title": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "title": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-15",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ant-form-item-control",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ant-form-item-children",
                  },
                  "children": Array [
                    Object {
                      "data": "mock-item",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "ant-form-extra",
                    },
                    "children": Array [
                      Object {
                        "data": "remind text",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "ant-form-extra",
                  },
                  "children": Array [
                    Object {
                      "data": "remind text",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "ant-form-item-children",
                    },
                    "children": Array [
                      Object {
                        "data": "mock-item",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-15",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ant-form-item-control",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ant-form-item-children",
                },
                "children": Array [
                  Object {
                    "data": "mock-item",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "ant-form-extra",
                  },
                  "children": Array [
                    Object {
                      "data": "remind text",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "ant-form-extra",
                },
                "children": Array [
                  Object {
                    "data": "remind text",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "ant-form-item-children",
                  },
                  "children": Array [
                    Object {
                      "data": "mock-item",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-9",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ant-form-item-required",
                "title": "mock-label",
              },
              "children": Array [
                Object {
                  "data": "mock-label",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "title": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "title": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
